students/DanCornutt/lessons/lesson07/database.py

```diff
@@ -0,0 +1,159 @@
+    """Returns dict of products available for rental"""
+    info = {}
+    qset = db['products'].find({"quantity_available": {"$gt": "0"}})
+    for item in qset:
+        id = item['product_id']
+        content = dict((k, item[k]) for k in (
+            'description', 'product_type', 'quantity_available'
+        ))
+        info.update({id: content})
+    return info
+
+
+def show_rentals(product_id):
+    """Returns a Python dictionary with the following user information from
+    users that have rented products matching product_id:
+    user_id.
+    name.
+    address.
+    phone_number.
+    email.
+    """
+    info = {}
+    qset = db.rentals.aggregate([
+        {"$match": {
+            "product_id": product_id
+        }},
+        {"$lookup": {
+            "localField": "user_id",
+            "from": "customers",
+            "foreignField": "user_id",
+            "as": "cust"
+        }},
+        {"$unwind": "$cust"},
+        {"$project": {
+            "user_id": 1,
+            "name": "$cust.name",
+            "address": "$cust.address",
+            "phone_number": "$cust.phone_number",
+            "email": "$cust.email"
+        }}
+    ])
+    for item in qset:
+        id = item['user_id']
+        content = dict((k, item[k]) for k in (
+            'name', 'address', 'phone_number', 'email'
+        ))
+        info.update({id: content})
+    return info
+
+
+class MyThread(Thread):
+
+    def __init__(self, directory, f_name, que):
+        Thread.__init__(self)
+        self.directory = directory
+        self.f_name = f_name
+        self.queue = que
+
+    def run(self):
+        """Reads csv by line. Uses csv dict iterator to loop through csv and add
+        document to collection
+        param1: directory
+        param2: csv file for collection
+        """
+
+        coll = self.f_name.split('.')[0]
+        cd = db[coll]
+        err = 0
+        with open(self.f_name, mode='r', encoding='utf-8-sig') as csv_f:
+            reader = csv.DictReader(csv_f)
+            for row in reader:
+                logging.debug("Parsing data: {}".format(row))
+                try:
+                    cd.insert_one(row)
+                except ValueError:
+                    err += 1
+        self.queue.put(cd.count())
+        self.queue.put(err)
+
+
+def import_data(dir_name, prod_f, cust_f, rent_f):
+    """
+    Fuction takes directory name and 3 csv files and then creates a new
+    MongoDB with 3 collections.
+    param1: directory name? I don't know the purpose
+    param2: products
+    param3: customers
+    param4: rentals
+    """
+    que = queue.Queue()
+
+    thread_p = MyThread(dir_name, prod_f, que)
+    thread_c = MyThread(dir_name, cust_f, que)
+    thread_r = MyThread(dir_name, rent_f, que)
+    thread_p.start()
+    thread_c.start()
+    thread_r.start()
+
+    for each in range(6):
+        thingy = que.get()
+        print ("This is the result: " + type(thingy) + str(thingy))
+
+    return que
```

Comment:
Nice threading. I love that your variable is named thingy. Looks like my code. ;-)

Reply:
thingy seem appropriate haha